### ISSUE TITLE
Ability to drag and drop breakpoints

### ISSUE THREAD

**Issue by ariffjeff** [microsoft/vscode#88227]
Issue Type: <b>Feature Request</b>

The ability to move breakpoints, especially those that are conditional, would be considerably easier than having to delete a breakpoint and then recreate it with the same condition information on another line. Drag and drop functionality would work pretty well here I imagine.

VS Code version: Code 1.41.1 (26076a4de974ead31f97692a0d32f90d735645c0, 2019-12-18T14:58:56.166Z)
OS version: Windows_NT x64 10.0.18362

**Comment by weinand** [comment]
yes, this would improve the usability of "rich" breakpoints.

**Comment by isidorn** [comment]
Agree, this is a fair feature request.

**Comment by ariffjeff** [comment]
As an added thought: For cases of files that are hundreds or thousands of lines long, it would probably be a good idea to be able to enter in a line number you want to move a breakpoint to instead of having to click and drag it over an entire document. The Breakpoints dropdown already displays the line numbers for each breakpoint so double clicking or right clicking them is perhaps the best way to enter line number values?

**Comment by joerohde** [comment]
This is becoming more frustrating as I use conditional breakpoints more and more.

It's a big break in flow to display/copy/re-create-condition-breakpoint/paste when I want to drag it down a line.

Also clicking or double clicking the line number in the breakpoints panel and having a context menu item for changing the line number in that panel would be a bonus.

**Comment by isidorn** [comment]
Overall this feature request makes sense, however this is a short iteration thus I do not think I will have time to tackle this milestone. Due to that pushing to May.

**Comment by isidorn** [comment]
This iteration is even shorter due to holidays. Thus pushing to on-deck and we would put this back on the plan when we have cycles.

**Comment by isidorn** [comment]
I have a simple version of this working https://github.com/microsoft/vscode/pull/114717
However for more powerfull support where we are able to render images while dragging and to be able to drag between multiple editors we would need this as a prerequest https://github.com/microsoft/vscode/issues/114776
Due to that moving this item to next milestone

**Comment by alefragnani** [comment]
Hi @isidorn ,

Are there any chances of exposing this _drag and drop_ capability/event to `Decorators` as well, so extensions could also subscribe?

Thank you

ps: I did search the open issues about it, but didn't find any 😬

**Comment by isidorn** [comment]
@alefragnani yes it would be possible, but we would first do the breakpoint in the core and only after that potential add this if there is enough extensions interested.. It would be best if you file a new issue. Thanks

**Comment by OneLookSigns** [comment]
@isidorn Further functionality could be to also allow Control + Drag so that breakpoints with conditional information can be duplicated to other lines.

**Comment by johncrim** [comment]
Could the scope of this be expanded to include logpoints? (IMO drag + drop is more valuable for conditional breakpoints and logpoints than standard breakpoints, due to the effort required today to change/move them). If yes, then I can close #182133 as a dup.

**Comment by jetpackguy** [comment]
Please make it happen 🙏

**Comment by MRogelio** [comment]
It has been over 4 years since this was opened. I'm wondering what's the issue and I get VSCode gets many requests as such, some way more complex and/or urgent

**Comment by kirkilj** [comment]
2025 checking in. This is my biggest annoyance on my current project. Ctrl-drag-n-drop to copy a conditional breakpoint would also be very useful. Another possible option in large files, besides entering a line number for the new breakpoint is to support context-menu item for copy and paste in the breakpoint bar.

**Comment by NathanDai5287** [comment]
> 2025 checking in. This problem makes me want to kms. Ctrl-drag-n-drop to copy a conditional breakpoint would also be very useful. Another possible option in large files, besides entering a line number for the new breakpoint is to support context-menu item for copy and paste in the breakpoint bar.

real

**Comment by edabor** [comment]
+1
